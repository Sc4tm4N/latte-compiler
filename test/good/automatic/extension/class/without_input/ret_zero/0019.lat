/* Should all types be legitly used. */

string getElem() {
   return "abba";
}

class Node {
  int elem;
  Node next;

  void setElem (int e)  { elem = e; }
  void setNext (Node n) { next = n; }

  int  getElem () { return elem; }
  Node getNext () { return next; }

  void test() {
    int a = self.getElem();
  }

  int test2() {
    return self.getElem();
  }

  string test3() {
    return getElem();
  }

  Node test4() {
    return getNext();
  }

  Node test5() {
    return self.getNext();
  }
}

int main() {
  Node n1 = new Node;
  Node n2 = new Node;
  n2.elem = 50;
  n2.next = (Node)null;
  n1.next = n2;
  n1.elem = 10;
  n1.test();
  string b = getElem();
  printString(b);
  int c = n1.test2();
  printInt(c);
  int d = n1.getElem();
  printInt(d);
  string e = n1.test3();
  printString(e);
  printInt(n1.test4().elem);
  printInt(n1.test5().elem);
  if(n1 == n1)
    return 0 ;
}
