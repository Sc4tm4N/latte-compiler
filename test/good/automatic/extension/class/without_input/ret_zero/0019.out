abba
10
10
abba
50
50
